src/main/scala/tech/cryptonomic/conseil/Conseil.scala

```diff
@@ -31,115 +42,157 @@ object Conseil
+            logger.error(
+              "The server was not started correctly, I failed to create the required Metadata service",
+              error
+            )
+            Await.ready(system.terminate(), 10.seconds)
+        }.flatMap(
+          runServer(_, server, platforms, securityApi, verbose)
+        )
 
-          onComplete(securityApi.validateApiKey(apiKey)).flatMap {
-            case Success(true) => provide(apiKey.getOrElse(""))
-            case _ =>
-              complete((Unauthorized, apiKey.fold("Missing API key") { _ =>
-                "Incorrect API key"
-              }))
+      sys.addShutdownHook {
+        serverBinding
+          .flatMap(_.unbind().andThen { case _ => logger.info("Server stopped...") })
+          .andThen {
+            case _ => system.terminate()
           }
+          .onComplete(_ => logger.info("We're done here, nothing else to see"))
       }
 
-      val tezosDispatcher = system.dispatchers.lookup("akka.tezos-dispatcher")
-
-      // This part is a temporary middle ground between current implementation and moving code to use IO
-      implicit val contextShift: ContextShift[IO] = IO.contextShift(executionContext)
-      val metadataCaching = MetadataCaching.empty[IO].unsafeRunSync()
-
-      lazy val transformation = new UnitTransformation(metadataOverrides)
-      lazy val cacheOverrides = new AttributeValuesCacheConfiguration(metadataOverrides)
-
-      lazy val tezosPlatformDiscoveryOperations =
-        TezosPlatformDiscoveryOperations(
-          ApiOperations,
-          metadataCaching,
-          cacheOverrides,
-          server.cacheTTL,
-          server.highCardinalityLimit
-        )(
-          executionContext,
-          contextShift
-        )
+  }
 
-      tezosPlatformDiscoveryOperations.init().onComplete {
-        case Failure(exception) => logger.error("Pre-caching metadata failed", exception)
-        case Success(_) => logger.info("Pre-caching successful!")
-      }
+  /** Reads configuration to setup the fundamental application services
+    * @param appConfig static configuration as read from files
+    * @return a metadata services object, if all went fine
+    */
+  def initServices(
+      server: ServerConfiguration,
+      platforms: PlatformsConfiguration,
+      metadataOverrides: MetadataConfiguration,
+      nautilusCloud: Option[NautilusCloudConfiguration]
+  )(implicit executionContext: ExecutionContext, system: ActorSystem, mat: ActorMaterializer): Try[MetadataService] = {
+
+    nautilusCloud.foreach { ncc =>
+      system.scheduler.schedule(ncc.delay, ncc.interval)(Security.updateKeys(ncc))
+    }
+
+    // This part is a temporary middle ground between current implementation and moving code to use IO
+    implicit val contextShift: ContextShift[IO] = IO.contextShift(executionContext)
+    val metadataCaching = MetadataCaching.empty[IO].unsafeRunSync()
+
+    lazy val transformation = new UnitTransformation(metadataOverrides)
```

Comment:
In hindsight, it's not obvious what UnitTransformation means in this context, just from looking at this line of code. Maybe a rename is in order for a different ticket?

Reply:
I's curious too, I guess the name is borrowed from math to mean that it will take an existing object and apply a transform that won't change the basic overall metrics (so to speak)?